### PR TITLE
add `chunk` func

### PR DESCRIPTION
**Pull request by Luoxin**

### CODE REVIEW

**Comment by elliotchance** [comment]
I'm happy to keep v1 on maintenance mode (forever), but any new functionality should only [go into v2](https://github.com/elliotchance/pie/tree/master/v2).

**Comment by Luoxin** [comment]
I do some work in v2，but I prefer sdk of v1

File: v2/chunk.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package pie
+
+import "fmt"
+
+// Chunk Split slice to chunks
+func Chunk[T any](ss []T, chunkSize int, callback func(ss []T) (stopped bool)) {
```
Comment (elliotchance) [inline v2/chunk.go:6]: This function feels like it might be ultra specific to your use case. Do you have some examples you can provide?

**Review by elliotchance — CHANGES_REQUESTED** [review]

**Comment by elliotchance** [comment]
Closing in favor of #184

**Pull request closed without merging**